env0/resource_cost_credentials_test.go

```diff
@@ -0,0 +1,394 @@
+package env0
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/env0/terraform-provider-env0/client"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestUnitAwsCostCredentialsResource(t *testing.T) {
```

Comment:
since all those tests are basically the same, and the only difference is the cloud provider
maybe something like 
https://github.com/env0/terraform-provider-env0/blob/3a532b76d558e0936d68ea9717e210e9ba6fe4bd/env0/resource_template_test.go#L367-L380

you can pass the relevant properties for each cloud provider and run all the test cases inside that loop?

Reply:
Will be more complex in my scenario if possible at all, the values for my tests are completely different, so the test step values, if its worth the effort Ill find a way to make it look better

Reply:
I don't want us for each new test to add it in 3 different places, this is wrong
so please find a way to DRY that

either by what I suggested (it might be bigger with a lot of values) or other solutions you can come up with